include/image-commands.mk

```diff
@@ -265,6 +265,13 @@ define Build/check-size
 	}
 endef
 
+define Build/check-size-firmware
+	sh $(TOPDIR)/scripts/check-size-firmware.sh $(KDIR)/image-$(firstword $(DEVICE_DTS)).dtb "$$(stat -c%s $@)" || { \
```

Comment:
Here, I'm still thinking about what's the best shape for the script:
- One could change the script to just be a get-size-firmware.sh, so the check would move to the Makefile here. This would enable use of the script in other places, but would also make handling of errors less comfortable (missing firmware node would be treated as "image too big" ...)
- One could move the stat command to the script and just pass $@ as second parameter here. This would make it more complex to just call the script to learn about firmware partition sizes...

Reply:
Indeed, it should be more generic, so it could be reused in other targets. Probably something like `get-dts-part-size.sh`.

> also make handling of errors less comfortable (missing firmware node would be treated as "image too big" ...)

We would use this script only if we know, that it's going to work with the DTS, right?

Reply:
> We would use this script only if we know, that it's going to work with the DTS, right?

Yes, I'm just not sure whether everyone will think of that.
However, it will fail when there is no firmware node, so the user will be notified. He just won't see the correct message then.